app/lib/task/global_lock.dart

```diff
@@ -41,8 +41,8 @@ class GlobalLock {
     final claimId = c._entry.claimId;
     var refreshed = Future.value(true);
     final done = Completer<void>();
-    try {
-      scheduleMicrotask(() async {
+    scheduleMicrotask(() async {
```

Comment:
This doesn't move try-catch into the microtask... it moves it out from `await refreshed` which is otherwise the only thing that could cause this.